tests/src/Interop/PInvoke/Attributes/LCID/LCIDTest.cs

```diff
@@ -0,0 +1,131 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+//The testcase focus test the BStr with embed null string
+using System.Runtime.InteropServices;
+using System;
+using System.Reflection;
+using System.Text;
+using CoreFXTestLibrary;
+
+class LCIDTest
+{
+    [DllImport(@"LCIDNative.dll", EntryPoint = "Marshal_InOut1")]
+    [LCIDConversionAttribute(0)]
+    [return: MarshalAs(UnmanagedType.LPStr)]
+    private static extern StringBuilder MarshalStrB_InOut1([In, Out][MarshalAs(UnmanagedType.LPStr)]StringBuilder s);
+
+    [DllImport(@"LCIDNative.dll", EntryPoint = "Marshal_InOut2")]
+    [LCIDConversionAttribute(1)]
+    [return: MarshalAs(UnmanagedType.LPStr)]
+    private static extern StringBuilder MarshalStrB_InOut2([In, Out][MarshalAs(UnmanagedType.LPStr)]StringBuilder s);
+
+    [DllImport(@"LCIDNative.dll", EntryPoint = "Marshal_InOut2", SetLastError = true)]
+    [LCIDConversionAttribute(1)]
+    [return: MarshalAs(UnmanagedType.LPStr)]
+    private static extern StringBuilder MarshalStrB_InOut3([In, Out][MarshalAs(UnmanagedType.LPStr)]StringBuilder s);
+
+    [DllImport(@"LCIDNative.dll", EntryPoint = "Marshal_InOut4", PreserveSig = false, SetLastError = true)]
+    [LCIDConversionAttribute(1)]
+    [return: MarshalAs(UnmanagedType.LPStr)]
+    private static extern StringBuilder MarshalStrB_InOut4([In, Out][MarshalAs(UnmanagedType.LPStr)]StringBuilder s);
```

Comment:
nit: could you give a good function name? try to avoid 1, 2, 3,4 .. such as MarshalStringBuilder_LCID_PreserveSig_SetLastError ?